packages/react/src/components/Notification/Notification.js

```diff
@@ -266,25 +267,33 @@ export function ToastNotification({
     [`${prefix}--toast-notification--${kind}`]: kind,
   });
 
+  const handleClose = useCallback(
+    (evt) => {
+      if (!onClose || onClose(evt) !== false) {
+        setIsOpen(false);
+      }
+    },
+    [onClose]
```

Comment:
I think we could have this be a function declaration versus `useCallback`. The rationale would be that the identify of `onClose` is likely to change between renders (if it's inline as an arrow function or a function declaration in a functional component). Since it is likely to change, then we are doing the work of equality check and re-creating the closure/function every render instead of just allocating the closure/function. Curious what your thoughts are 🤔